core/src/main/scala/tech/beshu/ror/accesscontrol/factory/BlockValidator.scala

```diff
@@ -54,8 +54,7 @@ object BlockValidator {
       .map(_.rule)
       .collect { case a: AuthenticationRule => a }
       .filter {
-        case _: GroupsOrRule => false
-        case _: GroupsAndRule => false
+        case _: GroupsRule[_] => false
```

Comment:
Here and in `UserDefinitionsDecoder` - the original implementation was created when there was only an "or" groups rule. I think that it is correct to match all groups rules here.